spec/controllers/users_controller_spec.rb

```diff
@@ -5,38 +5,50 @@
 describe Api::V1::UsersController do
   render_views
 
-  before(:each) do
-    @user = create :user
-    @friend = create :user
-  end
+  let!(:user) { create(:user) }
 
   describe "PUT 'update/:id'" do
-    it 'allows a user to be updated' do
-      @attr = { username: 'new username' }
-      request.headers['X-USER-TOKEN'] = @user.authentication_token
-      put :update, id: @user.id, user: @attr, format: 'json'
-      parsed_response = JSON.parse(response.body)
-      @user.reload
-      expect(response.response_code).to be 200
-      expect(@user.username).to eq @attr[:username]
+    let(:params) { { username: 'new username' } }
+
+    context 'with the correct data' do
+      it 'updates the user' do
+        request.headers['X-USER-TOKEN'] = user.authentication_token
+        put :update, id: user.id, user: params, format: 'json'
+
+        parsed_response = JSON.parse(response.body)
+        user.reload
+
+        expect(response.response_code).to be(200)
+        expect(user.username).to eq(params[:username])
+      end
     end
 
-    it 'should not allow to update a user (bad auth)' do
-      @attr = { username: 'new username' }
-      request.headers['X-USER-TOKEN'] = @user.authentication_token + 'wrong'
-      put :update, id: @user.id, user: @attr, format: 'json'
-      @user.reload
-      expect(response.status).to eq 401
-      expect(@user.username).to_not eq @attr[:username]
+    context 'with the incorrect auth' do
+      let(:params) { { username: 'new username' } }
+
+      it 'does not update the user' do
+        request.headers['X-USER-TOKEN'] = user.authentication_token + 'wrong'
+        put :update, id: user.id, user: params, format: 'json'
+
+        user.reload
+
+        expect(response.status).to eq(401)
+        expect(user.username).to_not eq(params[:username])
```

Comment:
You can remove `user.reload` in line 33 and include it in line 36 to avoid including extra lines.